src/Umbraco.Web.UI.Client/src/views/common/overlays/publish/publish.controller.js

```diff
@@ -1,23 +1,27 @@
 (function () {
     "use strict";
 
-    function PublishController($scope, $timeout) {
+    function PublishController($scope) {
 
         var vm = this;
-        vm.variants = $scope.model.variants;
+        var variants = $scope.model.variants;
         vm.changeSelection = changeSelection;
+        vm.loading = true;
+
+        vm.dirtyVariants = [];
```

Comment:
I've changed how this is implemented based on my previous comments in my branch using filters but the other issue with this is that you cannot publish a variant that you are currently editing if it's already published because 'dirty' in this case means that it has persisted pending changes but not active pending changes. I just wanted to mention this since it's a test case that could have been covered. I'll fix this up in my branch